crates/autopilot/src/run_loop.rs

```diff
@@ -212,20 +259,56 @@ impl RunLoop {
                 return;
             }
 
-            tracing::info!(driver = %driver.name, "settling");
-            let submission_start = Instant::now();
-            match self
-                .settle(driver, solution, auction_id, block_deadline)
+            let solved_order_uids: HashSet<_> = solution.orders().keys().cloned().collect();
+            self.in_flight_orders
+                .lock()
                 .await
-            {
-                Ok(()) => Metrics::settle_ok(driver, submission_start.elapsed()),
-                Err(err) => {
-                    Metrics::settle_err(driver, submission_start.elapsed(), &err);
-                    tracing::warn!(?err, driver = %driver.name, "settlement failed");
+                .extend(solved_order_uids.clone());
+
+            let solution_id = solution.id();
+            let driver_ = driver.clone();
+            let self_ = self.clone();
+
+            let existing_task = self.settlement_tasks.lock().await.remove(&driver_.name);
+            // If there was an existing task, wait for it to finish outside the lock scope.
+            if let Some(existing_task) = existing_task {
+                tracing::info!(driver = %driver_.name, "waiting for previous settlement task to finish");
+                if let Err(err) = existing_task.await {
```

Comment:
I'm not so sure about this logic here. This would mean if a solver wins back to back, we stop producing new auctions in parallele? That would be a bit unexpected for me.

I guess this was done in some way before, but can't we have a per solver queue with these settle attempts?

Reply:
But we still need to avoid situations where solutions accumulate faster than the settlement executes. With this approach, we will have at most one pending solution for the same solver(which was already computed in parallel).

Reply:
Why do we need to avoid such a situation? Settlements are timing out after a certain number of blocks, so we have a natural limit of how many solutions can be in the queue (I guess `n` where `n` is the block deadline).

I just find it very unintuitive that our level or parallelism should depend on who wins the auction.

Reply:
> I guess n where n is the block deadline

Yeah, I missed that actually. Thanks.